### ISSUE TITLE
secrets: uninitialized constant with rails 4.2.1 and devise 4.2.0

### ISSUE THREAD

**Issue by bithavoc** [plataformatec/devise#4186]
### Walk-around:
* Use devise **4.1.1**

### Problem:
`bundle exec rails generate devise:install` fails to run in a brand-new **rails 4.2.1** most likely due a missing `require('yaml')` somewhere.

### Stacktrace
```bash
➜  webpush git:(4-users) ✗ bundle exec rails generate devise:install
/Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/railties-4.2.1/lib/rails/application.rb:391:in `secrets': uninitialized constant Rails::Application::YAML (NameError)
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/devise-4.2.0/lib/devise/rails.rb:36:in `block in <class:Engine>'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `instance_exec'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `run'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/railties-4.2.1/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:228:in `block in tsort_each'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:347:in `each'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:347:in `call'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:347:in `each_strongly_connected_component'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:226:in `tsort_each'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/tsort.rb:205:in `tsort_each'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/railties-4.2.1/lib/rails/initializable.rb:54:in `run_initializers'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/railties-4.2.1/lib/rails/application.rb:352:in `initialize!'
	from /Users/bithavoc/Documents/Projects/webpush/config/environment.rb:5:in `<top (required)>'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/spring-1.7.2/lib/spring/application.rb:92:in `preload'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/spring-1.7.2/lib/spring/application.rb:143:in `serve'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/spring-1.7.2/lib/spring/application.rb:131:in `block in run'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/spring-1.7.2/lib/spring/application.rb:125:in `loop'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/spring-1.7.2/lib/spring/application.rb:125:in `run'
	from /Users/bithavoc/.rvm/gems/ruby-2.3.1/gems/spring-1.7.2/lib/spring/application/boot.rb:19:in `<top (required)>'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/bithavoc/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from -e:1:in `<main>'
```

Gemfile:
```rb
Gems included by the bundle:
  * actionmailer (4.2.1)
  * actionpack (4.2.1)
  * actionview (4.2.1)
  * activejob (4.2.1)
  * activemodel (4.2.1)
  * activerecord (4.2.1)
  * activesupport (4.2.1)
  * arel (6.0.3)
  * bcrypt (3.1.11)
  * binding_of_caller (0.7.2)
  * builder (3.2.2)
  * bundler (1.12.4)
  * byebug (9.0.5)
  * coffee-rails (4.1.1)
  * coffee-script (2.4.1)
  * coffee-script-source (1.10.0)
  * concurrent-ruby (1.0.2)
  * debug_inspector (0.0.2)
  * devise (4.1.1)
  * diff-lcs (1.2.5)
  * erubis (2.7.0)
  * execjs (2.7.0)
  * faker (1.5.0)
  * globalid (0.3.6)
  * i18n (0.7.0)
  * jbuilder (2.5.0)
  * jquery-rails (4.1.1)
  * json (1.8.3)
  * loofah (2.0.3)
  * mail (2.6.4)
  * mime-types (3.1)
  * mime-types-data (3.2016.0521)
  * mini_portile2 (2.1.0)
  * minitest (5.9.0)
  * multi_json (1.12.1)
  * nokogiri (1.6.8)
  * orm_adapter (0.5.0)
  * pg (0.18.4)
  * pkg-config (1.1.7)
  * rack (1.6.4)
  * rack-test (0.6.3)
  * rails (4.2.1)
  * rails-deprecated_sanitizer (1.0.3)
  * rails-dom-testing (1.0.7)
  * rails-html-sanitizer (1.0.3)
  * railties (4.2.1)
  * rake (11.2.2)
  * rdoc (4.2.2)
  * redis (3.3.0)
  * responders (2.2.0)
  * rspec-core (3.4.1)
  * rspec-expectations (3.4.0)
  * rspec-mocks (3.4.0)
  * rspec-rails (3.4.0)
  * rspec-support (3.4.1)
  * sass (3.4.22)
  * sass-rails (5.0.5)
  * sdoc (0.4.1)
  * spring (1.7.2)
  * sprockets (3.6.3)
  * sprockets-rails (3.1.1)
  * thor (0.19.1)
  * thread_safe (0.3.5)
  * tilt (2.0.5)
  * turbolinks (5.0.0)
  * turbolinks-source (5.0.0)
  * tzinfo (1.2.2)
  * uglifier (3.0.0)
  * warden (1.2.6)
  * web-console (2.3.0)
```

**Comment by lucasmazza** [comment]
This was (accidentally) fixed on Rails `4.2.3` by https://github.com/rails/rails/commit/64dacdd689ba3f3699f43d2971bbeb3a9d7195e9 - please upgrade your Rails dependency to `4.2.3` or later :)

**Comment by Helenkag** [comment]
How do I upgrade? I just installed Rails, shouldnt it be the latest? Thanks!
